File: 최윤경/전화번호 목록.js
/*
[접근]

1. 전화번호를 오름차순 정렬
2. 인접한 번호끼리 접두어 관계인지 확인
3. 접두어가 있으면 false, 없으면 true 반환
*/

function solution(phone_book) {
  var answer = true;

  phone_book.sort();

  for (let i = 0; i < phone_book.length - 1; i++) {
    if (phone_book[i + 1].startsWith(phone_book[i])) {
      return false;
    }
  }

  return true;
}

File: 표/6_hash_sort/전화번호목록.py
"""
어떤 번호가 다른 번호의 접두어인 경우가 있으면 false를 그렇지 않으면 true를 return
"""

def solution(phone_book):
    answer = True
    
    phone_book.sort()
    for i, p in enumerate(phone_book):
        if i == len(phone_book) - 1:
            return answer
        if phone_book[i+1].startswith(p):
            return False
    